Fluid/Filters/ColorFilters.cs

```diff
@@ -79,6 +79,44 @@ public static FluidValue ToHsl(FluidValue input, FilterArguments arguments, Temp
             }
         }
 
+        public static FluidValue ColorExtract(FluidValue input, FilterArguments arguments, TemplateContext context)
+        {
+            var value = input.ToStringValue();
+            HslColor hslColor;
+            RgbColor rgbColor;
+            if (HexColor.TryParse(value, out HexColor hexColor))
+            {
+                rgbColor = RgbColor.FromHex(hexColor);
+                hslColor = HslColor.FromHex(hexColor);
+            }
+            else if (RgbColor.TryParse(value, out rgbColor))
+            {
+                hslColor = HslColor.FromRgb(rgbColor);
+            }
+            else if (HslColor.TryParse(value, out hslColor))
+            {
+                rgbColor = RgbColor.FromHsl(hslColor);
+                // TODO: Check why 1 is added to RGB
+                rgbColor = new RgbColor(rgbColor.R - 1, rgbColor.G - 1, rgbColor.B - 1, rgbColor.A);
```

Comment:
Don't merge unless I fix a fractions issue that increases RGB by 1

Reply:
Convert.ToInt32?

Reply:
I already did, but I will try to check what's going on HSL to RGB algorithm